play.geforcenow.com - remove messages of no support

karlcow opened the pull request:
#### d9661a9bd3129fc92401e7715211fa504a6f2b87
<pre>
play.geforcenow.com - remove messages of no support
<a href="https://bugs.webkit.org/show_bug.cgi?id=303622">https://bugs.webkit.org/show_bug.cgi?id=303622</a>
<a href="https://rdar.apple.com/157575295">rdar://157575295</a>

Reviewed by Brent Fulgham.

nvidia which is the owner of play.geforcenow.com is warning users of
missing features in Safari. The features have now be added to WebKit.
(see the dependencies on <a href="https://webkit.org/b/303622">https://webkit.org/b/303622</a> )

This patch is a temporary quirk to support some internal automated
testing for a future release. We plan to remove this after our
colleagues at Nvidia are able to validate a developer beta and can
adjust their site to feature detect builds with the relevant features in
place. To remove the quirk see <a href="https://webkit.org/b/303718">https://webkit.org/b/303718</a>

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::needsGeforcenowWarningDisplayNoneQuirk const):
(WebCore::handleGeforcenowQuirks):
(WebCore::Quirks::determineRelevantQuirks):
* Source/WebCore/page/Quirks.h:
* Source/WebCore/page/QuirksData.h:
* Source/WebCore/style/StyleAdjuster.cpp:
(WebCore::Style::Adjuster::adjustForSiteSpecificQuirks const):

Canonical link: <a href="https://commits.webkit.org/304080@main">https://commits.webkit.org/304080@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b2c6ca7630ceb5d765d6cd07097d68b9f928d8c2

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/134523 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/6981 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/45752 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/142050 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/86491 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/6131f405-720e-4fa6-9274-8c7c51f39b58) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/136393 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/7584 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/6845 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/102816 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/86491 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/93e8b31f-2056-4152-8c04-93ae418ddbb8) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/137470 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/5300 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/120557 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/83612 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/abf48584-6c13-4fa5-b0e2-4be7891c93b6) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/5160 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/2776 "Passed tests") | [  ~~🛠 wpe-cairo-libwebrtc~~](https://ews-build.webkit.org/#/builders/166/builds/2665 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/114386 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/38697 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/144744 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/6659 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/39281 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/111217 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/6735 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/5585 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/111492 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/4994 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/116833 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/60519 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/20766 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/6712 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/35033 "Passed tests") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/157/builds/6530 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/70290 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/6763 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/6640 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebCore/page/Quirks.cpp:
```diff
@@ -2905,6 +2917,14 @@ static void handleEAQuirks(QuirksData& quirksData, const URL& /* quirksURL */, c
+    if (quirksDomainString != "play.geforcenow.com"_s) [[unlikely]]
+        return;
+
+    quirksData.enableQuirk(QuirksData::SiteSpecificQuirk::NeedsGeforcenowWarningDisplayNoneQuirk);
```
Let's add a FIXME that we will remove this when GeForceNow has time to update their site.

karlcow replied:
Done. Thanks for the review.

brentfulgham commented on Source/WebCore/page/Quirks.cpp:
```diff
@@ -721,6 +721,18 @@ bool Quirks::needsGoogleTranslateScrollingQuirk() const
+        return false;
+
+    return m_quirksData.quirkIsEnabled(QuirksData::SiteSpecificQuirk::needsGeforcenowWarningDisplayNoneQuirk);
+#else
```
We don't need this `#else` case (and it causes the build to fail! :-) )

karlcow replied:
The build is failing probably because of the wrong lowercase.

`needsGeforcenowWarningDisplayNoneQuirk` vs
`NeedsGeforcenowWarningDisplayNoneQuirk `

brentfulgham reviewed (approved):
@karlcow This looks good, but I think we should clarify in the ChangeLog that this is a temporary quirk being done to support internal testing at Apple, and not something we intend to ship to customers under the current plan.

karlcow committed:
play.geforcenow.com - remove messages of no support
https://bugs.webkit.org/show_bug.cgi?id=303622
rdar://157575295

Reviewed by Brent Fulgham.

nvidia which is the owner of play.geforcenow.com is warning users of
missing features in Safari. The features have now be added to WebKit.
(see the dependencies on https://webkit.org/b/303622 )

This patch is a temporary quirk to support some internal automated
testing for a future release. We plan to remove this after our
colleagues at Nvidia are able to validate a developer beta and can
adjust their site to feature detect builds with the relevant features in
place. To remove the quirk see https://webkit.org/b/303718

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::needsGeforcenowWarningDisplayNoneQuirk const):
(WebCore::handleGeforcenowQuirks):
(WebCore::Quirks::determineRelevantQuirks):
* Source/WebCore/page/Quirks.h:
* Source/WebCore/page/QuirksData.h:
* Source/WebCore/style/StyleAdjuster.cpp:
(WebCore::Style::Adjuster::adjustForSiteSpecificQuirks const):

Canonical link: https://commits.webkit.org/304080@main
